nemo_aligner/utils/train_script_utils.py

```diff
@@ -101,6 +102,7 @@ def init_using_ptl(ptl_trainer, ptl_model, train_dataloader, train_ds):
     disable_data_callbacks(ptl_model, train_dataloader, train_ds)
 
     call._call_setup_hook(ptl_trainer)
+    call._call_configure_model(ptl_trainer)
```

Comment:
For my education, does this fix anything? It looks like in [PTL](https://github.com/Lightning-AI/pytorch-lightning/blob/2524864b3cc8dae3552a4b5c3c819c2ce6f278af/src/lightning/pytorch/trainer/trainer.py#L950), this is indeed called after `_call_setup_hook`, but just wondering how we got away with not calling this

Reply:
Yes, this is very tricky. The PTL calls are 
```
- strategy.connect(model)
- strategy.setup_environment()
- call._call_setup_hook(self)
- call._call_configure_model(self)
- self.strategy.setup(self)
- _run_model [this is the actual training loop]
```

this function actually calls `configure_sharded_model` of the Megatron model which is a no-op if FSDP is off (check `megatron_base_model.py`). If FSDP is on, then this function calls `FullyShardedDataParallel(model, ...)` instead of the lightning trainer calling it. 

Without this function, FSDP does not work. I guess it worked so far because no one tested FSDP functionality in aligner? @JRD971000 told me none of the other aligner algos use fsdp. Is that true?

Reply:
Checked with Gerald and this change SG